### ISSUE TITLE
Getting CORS errors from a browser app client

### ISSUE THREAD

**Issue by AlGantori** [Quran-Tafseer/tafseer_api#63]
I am getting the following CORS error

> Access to XMLHttpRequest at 'http://api.quran-tafseer.com/tafseer/1/2/2' from origin 'http://localhost:8100' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.

![image](https://user-images.githubusercontent.com/11699920/111024440-a95fef80-8393-11eb-8639-b069e9563f5c.png)

I have downloaded the C#/WPF client and after a minor fix MainWindow.xaml.cs line 49 like so
```
QuranTafseer.Verse v = await client.GetVerseDetailsAsync(SuraNumbr, i);
```
it worked.  

However, I am developing a web-app using Ionic3.x/Angular5.x/TypeScript,  so I am afraid the CORS error is a browser specific issue which can only be fixed by correctly adjusting your server API configuration.

This is my understanding according to this documentation.
https://ionicframework.com/docs/troubleshooting/cors
![image](https://user-images.githubusercontent.com/11699920/111024421-94835c00-8393-11eb-9b26-80f799a8df84.png)

Similar documentation about CORS here
> https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS/Errors/CORSMissingAllowOrigin

and here
> https://enable-cors.org/server_apache.html

Personally, when I moved away from Godaddy and into my VPS/Linux, I saw CORS errors when trying to access static JSON from my own server.  My notes tells me I had to tweak some apache config with help from stackoverflow.

I have tried to derive a reproduction code using stackblitz, however, I get a different error...thus I was not able to reproduce the CORS errors.  Perhaps it's an artifact of running inside stackblitz.com site, nonetheless it too fails to retrieve your API data.

![image](https://user-images.githubusercontent.com/11699920/111024642-bf21e480-8394-11eb-999a-b551c93fcf2b.png)

You may find this code here and edit it if you wish.
https://stackblitz.com/edit/angular-x4qqet

Thank you for assisting, hopefully the sample code above could become another client sample against your API.

**Comment by AlGantori** [comment]
I found a simpler ERROR reproduction from simple HTML/JavaScript, you can simply put the following code into some index.html file and load it into the browser (Firefox or Chrome)

If you comment out the tafseer API URL and enable  "https://api.github.com"; you shall see success.

```
<!DOCTYPE html>
<html>

<head>
    <meta charset="utf-8" />
    <title>CORS-test</title>
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
    <script type='text/javascript'>
        function main() {
            console.log("main function");
            console.log("ajax request to the resource which will require cors enabled");
            var URL = "https://api.github.com";
            var URL = "http://api.quran-tafseer.com/tafseer/1/2/2";
            $.ajax
                ({
                    dataType: "json",
                    url: URL,
                    success: function (data) {
                        console.log("log response on success");
                        console.log(data);
                        alert(JSON.stringify(data))
                    }
                });
        }
    </script>
</head>

<body onLoad='main();'>
</body>

</html>
```

Even though I have average understanding of JS let alone ajax I tried to use JS alert to display the response/error
however, you will have to open the browser develop console with F12 to see the **CORS error**  apparently that logging is done by the browser network monitoring layer, according to this.

https://stackoverflow.com/questions/18864734/how-can-i-catch-jquery-ajax-errors

**Comment by EmadMokhtar** [comment]
@AlGantori The APIs do not support CORS. I will implement it and enable it soon. I will notify you once it is done.

**Comment by EmadMokhtar** [comment]
@AlGantori It is done. I tried your code and it is working.

**Comment by AlGantori** [comment]
Thank you. May Allah SWT reward you and your team with the very best.
I will start incorporating into my app.  I was about to develop some PHP on my server to act as the middle man.
This will save me from doing that.
If you read this let me know if you know of a Search API and if you have it on your roadmap?
Last night I did a quick sequential search on my uthmani json and the performance is fast enough.
I was wondering how an API would return highlighting for the strings search matches.

![image](https://user-images.githubusercontent.com/11699920/111912018-ff343780-8a24-11eb-8b3c-ccb65d317da1.png)
